Algorithm.CSharp/BaseBitcoin.cs

```diff
@@ -0,0 +1,87 @@
+﻿using NodaTime;
+using QuantConnect.Brokerages;
+using QuantConnect.Data.Market;
+using QuantConnect.Securities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+
+    /// <summary>
+    /// Base class for Bitcoin algorithms. This base class provides standard behaviour for trading BTCUSD on Bitfinex
+    /// trading on Bitfinex
+    /// </summary>
+    public abstract class BaseBitcoin : QCAlgorithm
```

Comment:
I've now modified the BaseBitcoin class [here](https://github.com/jameschch/Lean/commit/430f0396c571e0a4d318da10801ffa116c5023b3).

There are still some required initialization steps that might be better as changes to core classes. I'm sure your views on these kind of changes will be more informed than mine.